### PR TITLE
ONEUP-4193: Resizing bugfix

### PR DESCRIPTION
**Pull request by aramin**
# What
- Re-measure content width when marquee is resized.

# Why
- https://as-jira.axelspringer.de/browse/ONEUP-4193

### CODE REVIEW

File: src/Marquee.tsx [hunk 1]
```diff
@@ -188,24 +182,20 @@ export class Marquee extends Component<Props, State> {
...
+        });
 
-        window.addEventListener('resize', this.resizeCallback);
+        this.resizeCallback.observe(this.marqueeElement);
     }
 
     // This is to force a reflow which is necessary in order to transition styles when adding a class name.
-    private startAnimationAndForceReflow() {
+    private startAnimationAndForceReflow(): void {
         setTimeout(() => {
```
Comment (Sebastian-Fitzner) [inline src/Marquee.tsx:206]: Do you still need settimeout here?

File: src/__utils__/DimensionsHelper.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        DimensionsHelper.setAllProperties({
+            'offsetHeight': offsetHeight,
+            'offsetWidth': offsetWidth,
+            'innerWidth': innerWidth,
+            'innerHeight': innerHeight,
+        });
+    }
+
+    public restore(): void {
+        DimensionsHelper.setAllProperties({
```
Comment (Sebastian-Fitzner) [inline src/__utils__/DimensionsHelper.ts:47]: All the other methods are void as well. Let's be consistent and add the return type or remove it here.

File: src/Marquee.tsx [hunk 3]
```diff
@@ -106,6 +105,7 @@ export class Marquee extends Component<Props, State> {
                 <div
                     className={`preact-marquee__content ${this.state.animationClassName} ${this.state.pauseWhenHoveredClassName}`}
                     ref={this.saveContentReference}
```
Comment (Sebastian-Fitzner) [inline src/Marquee.tsx:108]: I think these ref functions are unnecessary. They will not be called anywhere else and they are not doing any additional things.

**Review by Sebastian-Fitzner — APPROVED** [review]

File: src/Marquee.tsx (hunk 3 above)
Comment (aramin) [reply src/Marquee.tsx:108]: it's just a callback to save the reference. what is the alternative?

File: src/Marquee.tsx (hunk 1 above)
Comment (aramin) [reply src/Marquee.tsx:206]: nope, works even without stopping and restarting the animation. will remove that functionallity.

File: src/Marquee.tsx (hunk 3 above)
Comment (Sebastian-Fitzner) [reply src/Marquee.tsx:108]: Just using the prop itself like `this.marqueeElement`... For now I do not see any benefit but we can also keep it.

**Review by Sebastian-Fitzner — APPROVED** [review]

**Pull request merged**